### PR TITLE
Docs: Clarified PR guidelines in maintainer guide

### PR DESCRIPTION
**Pull request by platinumazure**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain: Modifying PR guidelines in maintainer guide

**What changes did you make? (Give an overview)**

In the PR guide for maintainers:

* Separated human verification steps (check that commit tag is correct for issue) from bot verification steps (check that commit tag is present)
* Removed note about doc-only PRs not requiring an issue, since we only require issues for core changes
* Added human verification step to check that commit tag is correct for issue/problem
* Added human verification step to check for presence of issue, if PR is making a change to core
* Added note that team members can merge chores

**Is there anything you'd like reviewers to focus on?**

I tried to make changes that were either obvious clarifications/reorganizations, or that reflected our actual practices. Nonetheless, this might need TSC review to make sure I'm not inadvertently proposing to change maintainer guidelines.

### CODE REVIEW

**Comment by platinumazure** [comment]
**TSC Summary**: Goal is to clarify and improve the PR guidelines in the maintainer guide. I tried to make changes that were either obvious clarifications/reorganizations, or that reflected our actual practices. Nonetheless, this might need TSC review to make sure I'm not inadvertently proposing to change maintainer guidelines.

**TSC Question**: Are the proposed changes acceptable and accurate?

File: docs/maintainer-guide/pullrequests.md [hunk 1]
```diff
@@ -11,14 +11,14 @@ Anyone, both team members and the public, may leave comments on pull requests.
...
-1. Is the commit message summary in the correct format? Double-check that the tag ("Fix:", "New:", etc.) is correct based on the issue. Documentation-only pull requests do not require an issue.
-1. Does the commit summary reference an issue?
+1. Is the commit message summary in the correct format?
 1. Is the commit summary too long?
 
 The bot will add a comment specifying the problems that it finds. You do not need to look at the pull request any further until those problems have been addressed (there's no need to comment on the pull request to ask the submitter to do what the bot asked - that's why we have the bot!).
 
 Once the bot checks have been satisfied, you check the following:
 
+1. Double-check that the commit message tag ("Fix:", "New:", etc.) is correct based on the issue (or, if no issue is referenced, based on the stated problem).
```
Comment (kaicataldo) [inline docs/maintainer-guide/pullrequests.md]: Can we add something like` Note that documentation-only pull requests do not require an issue.` to the end of this?

**Comment by platinumazure** [comment]
That's no longer accurate (or rather sufficient): we only require issues on
core changes now. I believe I included a note to that effect elsewhere in
the commit but I'll double check.

**Comment by kaicataldo** [comment]
Sorry - you're right. This LGTM!

**Comment by btmills** [comment]
These changes were accepted by the TSC in the 2017-07-06 meeting.

**Pull request merged**